File: api/save_result.py
# api/save_result.py
import json
import os
from datetime import datetime

RESULT_DIR = "results"
os.makedirs(RESULT_DIR, exist_ok=True)

def handler(request):
    try:
        body = request.get("body")
        if isinstance(body, str):
            body = json.loads(body)

        username = body.get("username", "anonymous")
        score = body.get("score")
        topic = body.get("topic")
        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
        filename = f"{username}_{timestamp}.json"

        filepath = os.path.join(RESULT_DIR, filename)
        with open(filepath, "w", encoding="utf-8") as f:
            json.dump(body, f, ensure_ascii=False, indent=2)

        return {
            "statusCode": 200,
            "body": json.dumps({"message": "저장 완료", "file": filename})
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": f"에러 발생: {str(e)}"
        }

File: api/save_result.js
import fs from "fs";
import path from "path";

export default function handler(req, res) {
  try {
    if (req.method !== "POST") {
      return res.status(405).json({ error: "Method not allowed" });
    }

    const body = req.body;
    const username = body.username || "anonymous";
    const score = body.score;
    const topic = body.topic;

    const timestamp = new Date().toISOString().replace(/[:.]/g, "_");
    const filename = `${username}_${timestamp}.json`;

    const resultDir = path.join(process.cwd(), "results");
    if (!fs.existsSync(resultDir)) fs.mkdirSync(resultDir);

    const filepath = path.join(resultDir, filename);
    fs.writeFileSync(filepath, JSON.stringify(body, null, 2), "utf-8");

    res.status(200).json({ message: "저장 완료", file: filename });
  } catch (err) {
    res.status(500).send(`에러 발생: ${err.message}`);
  }
}
